stonesoup/movable/action/move_position_action.py

```diff
@@ -104,3 +104,105 @@ def __iter__(self):
+    )
+
+    n_samples: int = Property(
+        default=10,
+        doc="Number of samples to generate. This does not include the action "
+        "to remain at the current position, meaning :attr:`n_samples` +1 "
+        "actions will be generated."
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if self.action_space is not None:
+            if len(self.action_space) != len(self.action_mapping):
+                raise ValueError(f"Dimensions of action_space {self.action_space.shape} "
+                                 f"are not compatible with action_mapping of length "
+                                 f"{len(self.action_mapping)}. action_space should be "
+                                 f"of shape (ndim, 2) where ndim is the length of the "
+                                 f"action_mapping.")
+
+            if (np.any(self.current_value[self.action_mapping, :] < self.action_space[:, [0]])
+                    or np.any(self.current_value[self.action_mapping, :] > self.action_space[:, [1]])):  # noqa: E501
+                raise ValueError(f"Initial platform location {self.current_value} is not within "
+                                 f"the bounds of the action space {self.action_space}.")
+
+    @property
+    def default_action(self):
+        return MovePositionAction(generator=self,
+                                  end_time=self.end_time,
+                                  target_value=self.current_value)
+
+    def __contains__(self, item):
+        return item in iter(self)
+
+    @abstractmethod
+    def __iter__(self):
+        raise NotImplementedError
+
+
+class CircleSamplePositionActionGenerator(SamplePositionActionGenerator):
+    """Action generator which samples candidate future positions uniformly within
+    a circle around the current position. Circle radius is defined by the user
+    with :attr:`maximum_travel`. This generator is only applicable to 2D position
+    actions."""
+
+    maximum_travel: float = Property(
+        default=1.0,
+        doc="Maximum possible travel distance. Specifies the radius of "
+        "sampling area."
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if self.action_space is not None and len(self.action_space) != 2:
+            raise ValueError(f"Action mapping {self.action_mapping} does "
+                             f"not have 2 dimensions. "
+                             f":class:`~.CircleSamplePositionActionGenerator` "
+                             f"is designed for 2D action generation only.")
+
+    def __iter__(self):
+
+        yield MovePositionAction(generator=self,
+                                 end_time=self.end_time,
+                                 target_value=self.current_value)
+
+        generated_samples = 0
+
+        while generated_samples < self.n_samples:
+
+            radius_angle_sample = np.random.uniform([0, 0], [1, 2*np.pi], 2)
```

Comment:
This could be sampled outside of the while loop, something like: `np.random.uniform([0, 0], [1, 2*np.pi], (self.n_samples, 2)))`. Should be more efficient, especially when self.n_samples is large.